packages/lib/services/e2ee/cryptoTestUtils.ts

```diff
@@ -0,0 +1,286 @@
+import EncryptionService, { EncryptionMethod } from './EncryptionService';
+import BaseItem from '../../models/BaseItem';
+import Folder from '../../models/Folder';
+import MasterKey from '../../models/MasterKey';
+import Note from '../../models/Note';
+import Setting from '../../models/Setting';
+import shim from '../..//shim';
+import { getEncryptionEnabled, setEncryptionEnabled } from '../synchronizer/syncInfoUtils';
+
+interface DecryptTestData {
+	method: EncryptionMethod;
+	password: string;
+	plaintext: string;
+	ciphertext: string;
+}
+
+let serviceInstance: EncryptionService = null;
+
+// This is convenient to quickly generate some data to verify for example that
+// react-native-quick-crypto and node:crypto can decrypt the same data.
+export async function createDecryptTestData() {
+	const method = EncryptionMethod.StringV1;
+	const password = 'just testing';
+	const plaintext = '中文にっぽんご한국어😀\uD83D\0\r\nenglish01234567890';
+	const ciphertext = await serviceInstance.encrypt(method, password, plaintext);
+
+	return {
+		method: method,
+		password: password,
+		plaintext: plaintext,
+		ciphertext: ciphertext,
+	};
+}
+
+interface CheckTestDataOptions {
+	throwOnError?: boolean;
+	silent?: boolean;
+	testLabel?: string;
+}
+
+export async function checkDecryptTestData(data: DecryptTestData, options: CheckTestDataOptions = null) {
+	options = {
+		throwOnError: false,
+		silent: false,
+		...options,
+	};
+
+	// Verify that the ciphertext decrypted on this device and producing the same plaintext.
+	const messages: string[] = [];
+	let hasError = false;
+
+	try {
+		const decrypted = await EncryptionService.instance().decrypt(data.method, data.password, data.ciphertext);
+		if (decrypted !== data.plaintext) {
+			messages.push('Crypto Tests: Data could not be decrypted');
+			messages.push('Crypto Tests: Expected:', data.plaintext);
+			messages.push('Crypto Tests: Got:', decrypted);
+			hasError = true;
+		} else {
+			messages.push('Crypto Tests: Data could be decrypted');
+		}
+	} catch (error) {
+		hasError = true;
+		messages.push(`Crypto Tests: Failed to decrypt data: Error: ${error}`);
+	}
+
+	if (hasError && options.throwOnError) {
+		const label = options.testLabel ? ` (test ${options.testLabel})` : '';
+		throw new Error(`Testing Crypto failed${label}: \n${messages.join('\n')}`);
+	} else {
+		for (const msg of messages) {
+			if (hasError) {
+				console.warn(msg);
+			} else {
+				// eslint-disable-next-line no-console
+				if (!options.silent) console.info(msg);
+			}
```

Comment:
Consider using Joplin's `Logger` class here (e.g. `logger.warn`/`logger.debug`). On mobile, doing so should make the logged messages visible in Joplin's log screen (and eliminates the need for `eslint-disable-next-line`).

Reply:
I just reuse the code in `ppkTestUtils.ts`. Should I also change the `console.xx()` in it?